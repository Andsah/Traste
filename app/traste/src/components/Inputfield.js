import React from 'react';
import PropTypes from 'prop-types';
import {TextField} from '@mui/material';

/**
 * Custom TextField used for different data in the RenderPage.
 * @param {*} label Name of field.
 * @param {*} value Starting value.
 * @param {*} onChange Handler.
 * @param {*} error What will be shown for the error.
 * @param {*} type What type of values. Usally number or text.
 * @param {*} sx Styling
 * @return {TextField} Rendered field
 */
function Inputfield({label, value, onChange, error, type, sx}) {
  return (
    <TextField
      label={label}
      variant='outlined'
      sx={sx}
      value={value}
      type={type}
      onChange={onChange}
      inputProps={{'data-testid': 'inputfield'}}
      error={!!error}
      // helperText={error ? error.message : null}
    />
  );
}

Inputfield.defaultProps = {
  error: false,
  type: 'text',
  sx: {
    marginTop: '15px',
    backgroundColor: 'rgba(255,255,255,0.3)',
    width: '90vw',
  },
};

Inputfield.propTypes = {
  label: PropTypes.string.isRequired,
  value: PropTypes.any.isRequired,
  onChange: PropTypes.func.isRequired,
  error: PropTypes.any.isRequired,
  type: PropTypes.string.isRequired,
  sx: PropTypes.object,
};

export default Inputfield;
