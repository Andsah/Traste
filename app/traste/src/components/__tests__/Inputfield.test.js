import { red } from "@mui/material/colors";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import Inputfield from "../Inputfield";

afterEach(() => {
  cleanup();
});

test("Should render inputfield", () => {
  const propsData = { label: "Docket No.", name: "DocketNo", type: "string" };
  render(<Inputfield {...propsData} />);
  const inputElement = screen.getByTestId("inputfield");
  expect(inputElement).toBeInTheDocument();
  expect(inputElement).toHaveAccessibleName("Docket No.")
});

test("Should render inputfield", () => {
  const propsData = { label: "Docket No.", name: "DocketNo", type: "string" };
  render(<Inputfield {...propsData} />);
  const inputElement = screen.getByTestId("inputfield");
  expect(inputElement).toBeInTheDocument();
  expect(inputElement).not.toHaveAccessibleName("DocketNo.")
});

test("Matches snapshot", () => {
  const tree = renderer.create(<Inputfield />).toJSON();
  expect(tree).toMatchSnapshot();
});
