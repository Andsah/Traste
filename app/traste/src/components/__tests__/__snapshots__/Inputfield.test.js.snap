// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1hg5lel-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      data-testid="inputfield"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-nnbavb"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
